### PR TITLE
Fix get_area_slices

### PR DESCRIPTION
**Pull request by mraspaud**
- [x] Tests added 
 - [x] Tests passed 
 - [x] Passes ``git diff origin/develop **/*py | flake8 --diff`` 

This fixes https://github.com/pytroll/satpy/issues/218

### CODE REVIEW

File: pyresample/geometry.py [hunk 1]
```diff
@@ -1372,22 +1377,26 @@ def proj4_string(self):
...
-        if area_to_cover.proj_dict['proj'] == self.proj_dict['proj']:
+        if area_to_cover.proj_dict.get('proj') == self.proj_dict['proj']:
             logger.debug('Projections for data and slice areas are'
                          ' identical: %s', area_to_cover.proj_dict['proj'])
             # Get xy coordinates
             llx, lly, urx, ury = area_to_cover.area_extent
             x, y = self.get_xy_from_proj_coords([llx, urx], [lly, ury])
 
-            return slice(x[0], x[1] + 1), slice(y[1], y[0] + 1)
+            xstart = 0 if x[0] is np.ma.masked else x[0]
```
Comment (djhoese) [inline pyresample/geometry.py]: Are 'old style' area definitions the only ones that return a masked value? What do dask arrays return?

Comment (mraspaud) [reply pyresample/geometry.py]: `self.get_xy_from_proj_coords` doesn't produce a dask array, but I suppose that would return np.nan ?

File: setup.py [hunk 2]
```diff
@@ -33,7 +33,7 @@
                   'quicklook': ['matplotlib', 'cartopy', 'pillow'],
                   'dask': ['dask>=0.16.1']}
 
-test_requires = []
+test_requires = ['pytroll-schedule']
```
Comment (djhoese) [inline setup.py]: Should this maybe be added as an extra requirement too?

File: pyresample/geometry.py (hunk 1 above)
Comment (djhoese) [reply pyresample/geometry.py]: Time to test I guess 😉

File: setup.py (hunk 2 above)
Comment (mraspaud) [reply setup.py]: I'm looking at integrating the boundary stuff really soon, so I don't know

File: pyresample/geometry.py (hunk 1 above)
Comment (djhoese) [reply pyresample/geometry.py]: Or is it not necessary since you are passing the extents which are scalars?

Comment (mraspaud) [reply pyresample/geometry.py]: yes, exactly

File: pyresample/geometry.py [hunk 3]
```diff
@@ -1175,60 +1178,16 @@ def get_val(val, sub_val, max):
...
-                # Single element select
-                is_single_value = True
-
-                row_start = get_val(data_slice[0], 0, self.y_size)
-                col_start = get_val(data_slice[1], 0, self.x_size)
-
-                rows = 1
-                cols = 1
+            target_y = target_y[data_slice[0]]
+            target_x = target_x[data_slice[1]]
```
Comment (djhoese) [inline pyresample/geometry.py]: Wow this whole block of code got a lot smaller.

Comment (mraspaud) [reply pyresample/geometry.py]: Yes, but the downside is that the arange is called before the slicing is performed

Comment (djhoese) [reply pyresample/geometry.py]: Couldn't this be `da.arange`?

Comment (mraspaud) [reply pyresample/geometry.py]: Then we would need to put dask as a requirement.

Comment (djhoese) [reply pyresample/geometry.py]: Ah sorry, github's diff tool was making it look like this was the `_dask` method.

**Pull request merged**